### PR TITLE
table, tablets: use partitioned_sstable_set to collect compaction groups into a table

### PR DESCRIPTION
**Pull request by avikivity**
With the existing compound_sstable_set, every sstable is considered when reading a partition or when testing a partition for existence when merging a memtable into cache. Since there can be ~100 tablets on a shard, this results in CPU amplification of 100.

Fix by using partitioned_sstable_set, which filters out unrelated sstables.

It would be better to take advantage of the fixed boundaries between compaction groups, but this is a simple short-term fix that delivers most of the advantages.

### CODE REVIEW

File: replica/table.cc [hunk 1]
```diff
@@ -128,9 +128,13 @@ lw_shared_ptr<sstables::sstable_set> table::make_compound_sstable_set() {
     }
     // TODO: switch to a specialized set for groups which assumes disjointness across compound sets and incrementally read from them.
     // FIXME: avoid recreation of compound_set for groups which had no change. usually, only one group will be changed at a time.
-    auto sstable_sets = boost::copy_range<std::vector<lw_shared_ptr<sstables::sstable_set>>>(compaction_groups()
-        | boost::adaptors::transformed(std::mem_fn(&compaction_group::make_compound_sstable_set)));
-    return make_lw_shared(sstables::make_compound_sstable_set(schema(), std::move(sstable_sets)));
+    auto result = sstables::make_partitioned_sstable_set(schema(), false);
+    for (auto& cg : compaction_groups()) {
+        cg.main_sstables()->for_each_sstable([&] (sstables::shared_sstable sst) {
```
Comment (bhalevy) [inline replica/table.cc:133]: what about `maintenance_sstables`?
to be compatible with the existing compound_sstable_set you need `cg->make_compound_sstable_set()->for_each_sstable`

Comment (raphaelsc) [reply replica/table.cc:133]: Yes. Maybe maintenance set is today used during tablet streaming, even if no reshape is needed, and then sstables are moved to main set. It's something we have to revisit but not now.

**Comment by bhalevy** [comment]
This PR was superseded by https://github.com/scylladb/scylladb/pull/16918.
Closing...

**Pull request closed without merging**